internal/cmd/base/logging.go

```diff
@@ -59,77 +61,141 @@ func ProcessLogLevelAndFormat(flagLogLevel, flagLogFormat, configLogLevel, confi
 	return level, logFormat, nil
 }
 
+type grpcLogFakerLogHolder struct {
+	logger hclog.Logger
+}
+
+var grpcLogFaker *GRPCLogFaker
+
+func init() {
+	grpcLogFaker = &GRPCLogFaker{
+		log:    new(atomic.Bool),
+		logger: new(atomic.Pointer[grpcLogFakerLogHolder]),
+	}
+	grpclog.SetLoggerV2(grpcLogFaker)
+}
+
 type GRPCLogFaker struct {
-	Logger hclog.Logger
-	Log    bool
+	logger *atomic.Pointer[grpcLogFakerLogHolder]
+	log    *atomic.Bool
```

Comment:
Maybe name this something like `loggingEnabled` while we're here? It was a bit confusing to me to see `logger` and `log` in the same struct.
```suggestion
	loggingEnabled    *atomic.Bool
```

Reply:
Done.